src/StoreKeeper/WooCommerce/B2C/SegmentPriceProductBack.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace StoreKeeper\WooCommerce\B2C;
+
+use StoreKeeper\WooCommerce\B2C\Models\CustomerSegmentModel;
+use StoreKeeper\WooCommerce\B2C\Models\CustomerSegmentPriceModel;
+
+class SegmentPriceProductBack
+{
+    public function __construct()
+    {
+        add_filter('woocommerce_get_price_html', [$this, 'adjustProductPriceBasedOnQty'], 10, 2);
+    }
+
+    /**
+     * @return null
+     */
+    public static function getSegmentPrice($productId, $customerEmail, $qty)
```

Comment:
```suggestion
    public static function getSegmentPrice($productId, $userId, $qty)
```

I'm not sure why you search it by email